rasterio/tool.py

```diff
@@ -48,6 +49,71 @@ def stats(source):
     return Stats(numpy.min(arr), numpy.max(arr), numpy.mean(arr))
 
 
+def show_hist(source, bins=10, masked=True, title='Histogram'):
```

Comment:
Ah ok.  Right now we could do that with `Band()` objects but `DatasetReader()` and `RasterUpdater()` objects are not currently supported.  It's probably possible to look into the parent scope for `src`, but that seems kind of hacky and would only be guaranteed to work when using `rio insp`.

I'll try adding support for `DatasetReader()` and `RasterWriter()` so you could throw just about anything at this and get a figure.